pennylane/labs/resource_estimation/ops/qubit/non_parametric_ops.py

```diff
@@ -21,28 +21,94 @@
 
 
 class ResourceHadamard(qml.Hadamard, re.ResourceOperator):
-    """Resource class for the Hadamard gate."""
+    r"""Resource class for the Hadamard gate.
+
+    Resources:
+        The Hadamard gate is treated as a terminal gate and thus it cannot be decomposed
+        further. Requesting the resources of this gate raises a :code:`ResourcesNotDefined` error.
+
+    .. seealso:: :class:`~.Hadamard`
+
+    """
 
     @staticmethod
     def _resource_decomp(**kwargs) -> Dict[re.CompressedResourceOp, int]:
+        r"""Returns a dictionary representing the resources of the operator. The
+        keys are the operators and the associated values are the counts.
+
+        Resources:
+            The Hadamard gate is treated as a terminal gate and thus it cannot be decomposed
+            further. Requesting the resources of this gate raises a :code:`ResourcesNotDefined` error.
+        """
```

Comment:
Should have a Raises section. Also the other sections might want to be implemented with raising this error when appropriate as well.

Reply:
Added